src/entities/fractions/fraction.rs

```diff
@@ -57,21 +64,23 @@ where
     }
 
     // Returns the remainder after floor division as a new fraction
-    fn remainder(&self) -> Self {
+    fn remainder(&self) -> Result<Self, Error> {
         Self::new(
             self.numerator() % self.denominator(),
             self.denominator().clone(),
             self.meta(),
         )
+        .map_err(|err| Error::CreationError(format!("{}", err)))
```

Comment:
If you wish to map error to `CreationError` every time `new` is called, why don't you emit `CreationError` in `new`?

Reply:
true

Reply:
or i should remove it in general